webapp/src/ts/services/search-telemetry.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+
+import { TelemetryService } from '@mm-services/telemetry.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchTelemetryService {
+  private readonly CONTACT_SKIPPED_PROPERTIES = ['_id', '_rev', 'type', 'refid', 'geolocation'];
+  private readonly REPORT_SKIPPED_PROPERTIES = ['_id', '_rev', 'type', 'refid', 'content'];
+
+  constructor(
+    private readonly telemetryService: TelemetryService,
+  ) {}
+
+  public async recordContactSearch(contactDoc: Record<string, any>, search: string) {
+    const matchingProperties = await this.findMatchingProperties(contactDoc, search, this.CONTACT_SKIPPED_PROPERTIES);
+
+    for (const key of matchingProperties) {
```

Comment:
Minor, but is there any reason we need to do this sequentially instead of all at once via `Promise.all`?

Reply:
:+1:  In that case, seems reasonable to just do this however you feel is more readable.